### ISSUE TITLE
usage of sticky footer mixin

### ISSUE THREAD

**Issue by resistorsoftware** [Compass/compass#85]
Hi,

I am having trouble getting sticky footer to work. My screen.sass is set to http://gist.github.com/299438

My markup is basic and looks like: http://gist.github.com/299440

But when I review the page (Chrome)... there is a scroll bar and the footer is out of site. Could someone kindly point out any error I am making in using the mixin sticky-footer?

Thanks!!

**Comment by chriseppstein** [comment]
Is this a dup of http://github.com/chriseppstein/compass/issues/closed#issue/79

**Comment by resistorsoftware** [comment]
Not sure... The sticky footer has caused me enough problems to force me to try another approach. At the moment, even though it mostly works, some content is clearly rendering into the footer now, and that is just not nice...

**Comment by chriseppstein** [comment]
Would love to know how we can make it better from either a docs or bug-fix perspective. Any information you can provide is helpful.

**Comment by joshjhall** [comment]
Here's something I've been fiddling with and using in several projects.  It seems to serve my purposes well and works in most browsers (at least I haven't found anything major that it blows up in yet).  I also like how it keeps the markup simple and easily integrates with Blueprint.

Thoughts?

http://gist.github.com/316629

**Comment by chriseppstein** [comment]
The compass core framework cannot have any dependencies on the other frameworks. But we could have a grid-based sticky footer than depends on the compass core. This isn't the same output and so it may not actually work for you, but it is in line with how I'd like to see the core abstractions being built upon.

http://gist.github.com/316852

**Comment by joshjhall** [comment]
Fair point about keeping the compass core away from dependencies.  I thought about that over the weekend, but haven't been near a computer the last couple days.

I noticed that the auto left/right margins aren't really necessary if we're using the blueprint +container call, which simplifies some things.

I also noticed there is a bug in the sticky footer implementation in IE8.  The #{root_selector} should also have ":height auto !important"... without this IE has a tendency of letting the footer push up into the content.  The #{root_selector} also needs ":height 100%" for IE6 to behave correctly.

In fact, after reviewing this again, I think the existing sticky_footer will work fine with grid frameworks with those patches.  I'd be happy to write up a few examples of how to use it in more advanced situations (to the best of my ability at least).

http://gist.github.com/316629
